src/vivarium/framework/components/parser.py

```diff
@@ -180,85 +152,84 @@ def _process_level(level, prefix):
 
         component_list = []
         for name, child in level.items():
-            component_list.extend(_process_level(child, prefix + [name]))
+            component_list.extend(self._process_level(child, prefix + [name]))
 
         return component_list
 
-    return _process_level(component_config, [])
+    def prep_components(
+        self, component_list: Union[List[str], Tuple[str]]
+    ) -> List[Tuple[str, Tuple]]:
+        """Transform component description strings into tuples of component paths
+        and required arguments.
 
+        Parameters
+        ----------
+        component_list
+            The component descriptions to transform.
 
-def prep_components(component_list: Union[List[str], Tuple[str]]) -> List[Tuple[str, Tuple]]:
-    """Transform component description strings into tuples of component paths
-    and required arguments.
+        Returns
+        -------
+        List[Tuple[str, Tuple]]
+            List of component/argument tuples.
+        """
+        components = []
+        for c in component_list:
+            path, args_plus = c.split("(")
+            cleaned_args = self.clean_args(args_plus[:-1].split(","), path)
+            components.append((path, cleaned_args))
+        return components
+
+    @staticmethod
```

Comment:
Also...why use a static method? I'm still uncomfortable when/where to use these. In this case I see how the cleaning method should not be specific to each instance, but it also seems to be the case for `prep_components` above which you did not specify as static.

Reply:
This article does a good job of explaining when to use them (and class methods): https://realpython.com/instance-class-and-static-methods-demystified/